### PR TITLE
[Merged by Bors] - feat(Algebra/GroupPower/Basic): add zpow_eq_zpow_zmod

### PR DESCRIPTION
**Pull request by kbuzzard**
To match `pow_eq_pow_mod` for monoids.

---

Also beautify the proof of `pow_eq_pow_mod` (thanks Riccardo!) and add a hybrid `zpow_eq_zpow_mod`.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/GroupPower/Basic.lean [hunk 1]
```diff
@@ -476,6 +476,15 @@ lemma mul_zpow_self (a : G) (n : ℤ) : a ^ n * a = a ^ (n + 1) := (zpow_add_one
 #align zpow_mul_comm zpow_mul_comm
 #align zsmul_add_comm zsmul_add_comm
 
+theorem zpow_eq_zpow_zmod {x : G} (m : ℤ) {n : ℤ} (h : x ^ n = 1) :
```
Comment (ericrbg) [inline Mathlib/Algebra/GroupPower/Basic.lean:479]: I'm not sure if zmod is the right name for this, but I don't have any other suggestions either.

Comment (riccardobrasca) [reply Mathlib/Algebra/GroupPower/Basic.lean:479]: This `%` is [Int.mod](https://leanprover-community.github.io/mathlib4_docs/Init/Data/Int/Basic.html#Int.mod) right? In this case the convention seems to just use `mod`.

**Comment by riccardobrasca** [comment]
Thanks!

bors d+

File: Mathlib/Algebra/GroupPower/Basic.lean (hunk 1 above)
Comment (kbuzzard) [reply Mathlib/Algebra/GroupPower/Basic.lean:479]: Oh thanks! In which case maybe I'll go for `mod` and `mod'`.

File: Mathlib/Algebra/GroupPower/Basic.lean [hunk 2]
```diff
@@ -476,14 +476,14 @@ lemma mul_zpow_self (a : G) (n : ℤ) : a ^ n * a = a ^ (n + 1) := (zpow_add_one
 #align zpow_mul_comm zpow_mul_comm
 #align zsmul_add_comm zsmul_add_comm
 
-theorem zpow_eq_zpow_zmod {x : G} (m : ℤ) {n : ℤ} (h : x ^ n = 1) :
+theorem zpow_eq_zpow_mod {x : G} (m : ℤ) {n : ℤ} (h : x ^ n = 1) :
     x ^ m = x ^ (m % n) := by
 calc
     x ^ m = x ^ (m % n + n * (m / n)) := by rw [Int.emod_add_ediv]
     _ = x ^ (m % n) := by simp [zpow_add, zpow_mul, h]
```
Comment (riccardobrasca) [inline Mathlib/Algebra/GroupPower/Basic.lean:483]:
```suggestion
theorem zpow_eq_zpow_mod {x : G} (m : ℤ) {n : ℤ} (h : x ^ n = 1) :
    x ^ m = x ^ (m % n) :=
  calc
    x ^ m = x ^ (m % n + n * (m / n)) := by rw [Int.emod_add_ediv]
    _ = x ^ (m % n) := by simp [zpow_add, zpow_mul, h]
```

**Review by riccardobrasca — COMMENTED** [review]
It seems that it actually uses [Int.emod](https://leanprover-community.github.io/mathlib4_docs/Std/Data/Int/Basic.html#Int.emod), so you may want to check it is the version you want (and change the name)...

**Comment by riccardobrasca** [comment]
This seems very confusing, we both have [Int.instModInt](https://leanprover-community.github.io/mathlib4_docs/Init/Data/Int/Basic.html#Int.instModInt) and [Int.instModInt_1](https://leanprover-community.github.io/mathlib4_docs/Std/Data/Int/Basic.html#Int.instModInt_1), and I am afraid `%` uses one or the other depending on who know what.

**Comment by kbuzzard** [comment]
Wooah yeah it does seem to be `emod`! I don't want this lemma in practice, I only need the primed corollary (when we're dividing by an int coerced from nat), and my recollection is that there the two choices are the same. I've changed the names to emod. Well spotted :-)

**Pull request closed without merging**